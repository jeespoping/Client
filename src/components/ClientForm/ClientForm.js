import React, { useState } from "react";
import { Button, Form, Input, Select } from "semantic-ui-react";
import "./ClientForm.scss";

export default function ClientForm() {
  const [active, setActive] = useState(false);
  const [inputValue, setInputValue] = useState("");
  const [selectValue, setSelectValue] = useState("");

  const options = [
    {
      key: "CC",
      text: "Cédula de ciudadanía",
      value: "CC",
    },
    {
      key: "TI",
      text: "Tarjeta de identidad",
      value: "TI",
    },
    {
      key: "CE",
      text: "Cédula Extrangera",
      value: "CE",
    },
  ];

  const handleChange = (e) => {
    const number = e.target.value
      .replace(",", "")
      .replace(/\B(?=(\d{3})+(?!\d))/g, ",");

    setInputValue(number);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
  };

  return (
    <div className="client-form">
      <p>Todos los campos son obligatorios</p>

      <Form>
        <Form.Field>
          <label>Tipo de documento</label>
          <Select
            onChange={(_, data) => setSelectValue(data)}
            name="type"
            value={selectValue}
            options={options}
          />
        </Form.Field>
        <Form.Field>
          <label>Número de documento</label>
          <Input name="number" value={inputValue} onChange={handleChange} />
        </Form.Field>
        <Button type="submit" active={active}>
          Buscar
        </Button>
      </Form>
    </div>
  );
}
