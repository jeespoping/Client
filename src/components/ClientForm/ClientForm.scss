@import "../../SCSS/index.scss";

.client-form{
    position: relative;
    background-color: $background-light;
    width: 500px;
    padding: 0px 50px 50px 50px;
    border-radius: 10px;

    p{
        &:first-of-type{
            color: $font-light-gray;
        }
        
    }

    .ui.form{
        margin-top: 20px;
    }

    .button{
        display: flex;
        flex-direction: column;
        justify-content: center;
        align-items: center;

        .ui.button{
            background-color: $background-grey;
            color: $font-light;
            padding: 10px 60px;
            border-radius: 10px;
        }
    }
}