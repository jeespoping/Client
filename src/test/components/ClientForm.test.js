import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import ClientForm from "../../components/ClientForm/ClientForm";

describe("Prueba en <App />", () => {
  let wrapper = shallow(<ClientForm />);

  test("Debe mostrar correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("El boton debe de estar desactivado", () => {
    const button = wrapper.find("Button");

    expect(button.prop("disabled")).toBe(true);
  });
});
