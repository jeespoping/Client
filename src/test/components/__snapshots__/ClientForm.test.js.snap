// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <App /> Debe mostrar correctamente 1`] = `
<div
  className="client-form"
>
  <p>
    Todos los campos son obligatorios
  </p>
  <Form
    as="form"
    onSubmit={[Function]}
  >
    <FormField>
      <label>
        Tipo de documento
      </label>
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "CC",
              "text": "Cédula de ciudadanía",
              "value": "CC",
            },
            Object {
              "key": "TI",
              "text": "Tarjeta de identidad",
              "value": "TI",
            },
            Object {
              "key": "CE",
              "text": "Cédula Extrangera",
              "value": "CE",
            },
          ]
        }
        value=""
      />
    </FormField>
    <FormField>
      <label>
        Número de documento
      </label>
      <FormInput
        as={[Function]}
        control={[Function]}
        name="number"
        onChange={[Function]}
        value=""
      />
    </FormField>
    <div
      className="button"
    >
      <Button
        as="button"
        disabled={true}
        type="submit"
      >
        Buscar
      </Button>
    </div>
  </Form>
</div>
`;
